lib/base/registry.hpp

```diff
@@ -44,33 +57,36 @@ class Registry
 
 	ItemMap GetItems() const
 	{
-		std::unique_lock<std::mutex> lock(m_Mutex);
+		auto lock (ReadLockUnlessFrozen());
 
 		return m_Items; /* Makes a copy of the map. */
 	}
 
-	boost::signals2::signal<void (const String&, const T&)> OnRegistered;
-	boost::signals2::signal<void (const String&)> OnUnregistered;
+	/**
+	 * Freeze the registry, preventing further updates.
+	 *
+	 * This only prevents inserting, replacing or deleting values from the registry.
+	 * This operation has no effect on objects referenced by the values, these remain mutable if they were before.
+	 */
+	void Freeze()
+	{
+		std::shared_lock lock (m_Mutex);
```

Comment:
Why do you need lock this here? Isn't `m_Frozen` atomic?

Reply:
* #9627 also kept the lock in Freeze() 🤷‍♂️

Reply:
* #10414 also kept the locks in Freeze()!